plugins/registry/handler.go

```diff
@@ -282,6 +312,21 @@ func writeStatusInternalError(req *http.Request, w http.ResponseWriter) {
 	}
 }
 
+func writeStatusNotFoundError(req *http.Request, w http.ResponseWriter, error string) {
+	if req.ProtoMajor == 2 && strings.Contains(req.Header.Get("Content-Type"), "application/grpc") {
+		w.Header().Set("Content-Type", "application/grpc")
+		w.Header().Set("Trailer", "Grpc-Status")
+		w.Header().Add("Trailer", "Grpc-Message")
+		w.Header().Set("Grpc-Status", "5")
+		w.Header().Set("Grpc-Message", "NOT_FOUND")
+	} else {
+		w.WriteHeader(http.StatusNotFound)
```

Comment:
Note that `NotFound` will cause a retry loop while other statuses will fail immediately. Perhaps we can use `FAILED_PRECONDITION` (9) instead

Reply:
I didn't realize that, TIL 🙇🏼 , will update the status code